pkg/operator/encryption/prune_controller_test.go

```diff
@@ -0,0 +1,235 @@
+package encryption
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	clientgotesting "k8s.io/client-go/testing"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+func TestEncryptionPruneController(t *testing.T) {
+	scenarios := []struct {
+		name                     string
+		initialSecrets           []*corev1.Secret
+		encryptionSecretSelector metav1.ListOptions
+		targetNamespace          string
+		targetGRs                map[schema.GroupResource]bool
+		// expectedActions holds actions to be verified in the form of "verb:resource:namespace"
+		expectedActions       []string
+		expectedEncryptionCfg *apiserverconfigv1.EncryptionConfiguration
+		validateFunc          func(ts *testing.T, actions []clientgotesting.Action, initialSecrets []*corev1.Secret)
+	}{
+		{
+			name:            "no-op only 10 keys were migrated",
+			targetNamespace: "kms",
+			targetGRs: map[schema.GroupResource]bool{
+				schema.GroupResource{Group: "", Resource: "secrets"}: true,
+			},
+			initialSecrets: func() []*corev1.Secret {
+				ns := "kms"
+				all := []*corev1.Secret{}
+				all = append(all, createMigratedEncryptionKeySecretsWithRndKey(t, 10, ns, "secrets")...)
+				all = append(all, createEncryptionKeySecretWithRawKey(ns, nil, 11, []byte("cfbbae883984944e48d25590abdfd300")))
+				return all
+			}(),
+			expectedActions: []string{"list:pods:kms", "get:secrets:kms", "list:secrets:openshift-config-managed", "list:secrets:openshift-config-managed"},
+		},
+
+		{
+			name:            "14 keys were migrated, 1 of them is used, 10 are kept, 3 most recent are pruned",
+			targetNamespace: "kms",
+			targetGRs: map[schema.GroupResource]bool{
+				schema.GroupResource{Group: "", Resource: "secrets"}: true,
+			},
+			initialSecrets: createMigratedEncryptionKeySecretsWithRndKey(t, 14, "kms", "secrets"),
+			expectedActions: []string{
+				"list:pods:kms",
+				"get:secrets:kms",
+				"list:secrets:openshift-config-managed",
+				"list:secrets:openshift-config-managed",
+				"update:secrets:openshift-config-managed",
+				"delete:secrets:openshift-config-managed",
```

Comment:
it would be nice if we could include the name of the deleted resource.

Reply:
follow-up. Wasn't that your test? :-)